ray-operator/controllers/ray/common/job.go

```diff
@@ -54,96 +54,136 @@ func GetMetadataJson(metadata map[string]string, rayVersion string) (string, err
 	return pkgutils.ConvertByteSliceToString(metadataBytes), nil
 }
 
-// GetK8sJobCommand builds the K8s job command for the Ray job.
-func GetK8sJobCommand(rayJobInstance *rayv1.RayJob) ([]string, error) {
-	address := rayJobInstance.Status.DashboardURL
+// BuildJobSubmitCommand builds the job submit command based on submission mode.
+func BuildJobSubmitCommand(rayJobInstance *rayv1.RayJob, submissionMode rayv1.JobSubmissionMode) ([]string, error) {
+	var address string
+	var waitLoop []string
+
+	switch submissionMode {
+	case rayv1.SidecarMode:
+		// Sidecar submitter talks to the local head Pod dashboard.
+		address = "http://127.0.0.1:8265"
+		// Wait until dashboard is reachable before proceeding.
+		waitLoop = []string{
```

Comment:
Can you use something similar to the Ray head readiness probe https://github.com/ray-project/kuberay/blob/edf7de8d6abff398fc989e5c33eb264e2d2015cb/ray-operator/controllers/ray/common/pod.go#L347?